greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/DeleteCommand.java

```diff
@@ -41,6 +41,11 @@ protected void doProcess(ImapRequestLineReader request,
                 folder.getFullName().equals(session.getSelected().getFullName())) {
             session.deselect();
         }
+
+        if (mailboxName.toLowerCase().equals("inbox")) {
```

Comment:
I suggest moving this to `ImapHostManagerImpl` too. You might reuse the check for the INBOX folder as in `renameMailbox`.

Reply:
Moved it to `ImapHostManagerImpl`